src/ImageSharp/Formats/Gif/GifDecoder.cs

```diff
@@ -38,9 +38,10 @@ public Image<TPixel> Decode<TPixel>(Configuration configuration, Stream stream)
             {
                 Size dims = decoder.Dimensions;
 
-                // TODO: use InvalidImageContentException here, if we decide to define it
-                // https://github.com/SixLabors/ImageSharp/issues/1110
-                throw new ImageFormatException($"Can not decode image. Failed to allocate buffers for possibly degenerate dimensions: {dims.Width}x{dims.Height}.", ex);
+                GifThrowHelper.ThrowInvalidImageContentException($"Can not decode image. Failed to allocate buffers for possibly degenerate dimensions: {dims.Width}x{dims.Height}.", ex);
+
+                // Not reachable, as the previous statement will throw a exception.
+                return null;
```

Comment:
@JimBobSquarePants is there a way to annotate a method, that it will always throw to avoid this return statement?

Reply:
Unfortunately not. There's something coming in NET5 I believe but it won't be backwards compatible.